File: GOL-C++/matrix/Mlecz.cpp
#include <iostream>

#include "Swiat.h"
#include "Roslina.h"
#include "Mlecz.h"

Mlecz::Mlecz() : Roslina() {
	sila = 0;
}

Mlecz::Mlecz(int x, int y, Swiat* swiat) : Roslina(0, x, y, swiat) {}

char Mlecz::GetAwatar() const {
	return awatar;
}

string Mlecz::GetGatunek() const {
	return "Mlecz";
}

void Mlecz::akcja() {
	int i = 0;
	while (i < 3) {
		int ran = (rand() % 100) + 1;
		if (ran < ratio)
			Roslina::akcja();
		i++;
	}
}

void Mlecz::kolizja(Organizm* atakujacy) {
	Roslina::kolizja(atakujacy);
}

File: GOL-Java/Mlecz.java
package pl.edu.pg.eti.ksg.po.project2.rosliny;

import pl.edu.pg.eti.ksg.po.project2.Roslina;
import pl.edu.pg.eti.ksg.po.project2.Swiat;
import pl.edu.pg.eti.ksg.po.project2.Punkt;

import java.util.Random;
import java.awt.*;

public class Mlecz extends Roslina {
    private static final int sila = 0;
    private static final int inicjatywa = 0;

    public Mlecz(Swiat swiat, Punkt pozycja, int turaUrodzenia) {
        super(TypOrganizmu.MLECZ, swiat, pozycja,
                turaUrodzenia, sila, inicjatywa);
        setKolor(Color.YELLOW);
        setSzansaRozmnazania(0.2);
    }

    @Override
    public void Akcja() {
        Random rand = new Random();
        for (int i = 0; i < 3; i++) {
            int tmpLosowanie = rand.nextInt(100);
            if (tmpLosowanie < getSzansaRozmnazania()) Rozprzestrzenianie();
        }
    }

    @Override
    public String GetGatunek() {
        return "Mlecz";
    }
}
